tests/manifest/test_generator.py

```diff
@@ -0,0 +1,64 @@
+from unittest import TestCase
+
+from ingest.api.ingestapi import IngestApi
+from mock import MagicMock
+
+from manifest.generator import ManifestGenerator
+from manifest.manifests import ProcessInfo
+from tests.mocks.ingest import MockIngestAPI
+
+
+def process_info_converter(process_info: dict) -> ProcessInfo:
+    convert = ProcessInfo()
+    convert.project = process_info.get('project')
+    convert.input_biomaterials = process_info.get('input_biomaterials')
+    convert.derived_by_processes = process_info.get('derived_by_processes')
+    convert.input_files = process_info.get('input_files')
+    convert.derived_files = process_info.get('derived_files')
+    convert.protocols = process_info.get('protocols')
+    convert.supplementary_files = process_info.get('supplementary_files')
+    convert.input_bundle = process_info.get('input_bundle')
+    return convert
+
+
+class TestGenerator(TestCase):
+    def setUp(self) -> None:
+        self.maxDiff = None
+        self.ingest = MagicMock(spec=IngestApi, wraps=MockIngestAPI())
+
+    def test_get_all_process_info(self):
+        # given:
+        generator = ManifestGenerator(ingest_api=self.ingest)
+        assay_process = self.ingest.get_entity_by_uuid('processes', 'mock-assay-process')
+        example_process_info = self.ingest.get_entity_by_uuid('processes', 'example-process-info')
```

Comment:
It looks like we're using/hijacking the mock's `get_entity_by_uuid()` more as a test utility method here, rather than a mock (using it to read a JSON-ified ProcessInfo() rather than truly getting an entity by its UUID). I see how it works, and the convenience of it, but it looks confusing to have the same method retrieve an ingest entity in one instance, but then a ProcessInfo() in another.

We could instead have a TestUtil() class that reads and returns these JSON files. The TestUtil() could be used directly here to fetch the test JSON-ified ProcessInfo(), and also be used by the MockIngestApi() for fetching mock entities from disk.